### PR TITLE
Update copyright for all files that LANL employees touched in 2021

### PR DESCRIPTION
**Pull request by AndrewGaspar**
## PR Summary

Fails the build if LANL employees don't do this.

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md

### CODE REVIEW

**Review by Yurlungur — APPROVED** [review]

File: cmake/CheckCopyrightScript.cmake [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#=========================================================================================
+
+# This script checks that a file has an up to date copyright. Only runs for
+# users with @lanl.gov email addresses.
+
+set(SOURCE_DIR ${CMAKE_ARGV3})
+set(GIT_EXECUTABLE ${CMAKE_ARGV4})
+set(INPUT ${CMAKE_ARGV5})
```
Comment (JoshuaSBrown) [inline cmake/CheckCopyrightScript.cmake:19]: What does INPUT do?

File: cmake/CheckCopyrightScript.cmake [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+string(TIMESTAMP CURRENT_YEAR "%Y")
+
+# Check if the file has an up to date copyright
+file(READ ${SOURCE_DIR}/${INPUT} CONTENTS)
+string(REGEX MATCH "\\(C\\) \\(or copyright\\) 2020-${CURRENT_YEAR}\\. Triad National Security, LLC\\. All rights reserved\\." HAS_COPYRIGHT ${CONTENTS})
```
Comment (JoshuaSBrown) [inline cmake/CheckCopyrightScript.cmake]: So if I add a new file this year to Parthenon is the copyright still supposed to start from 2020 for that file?

File: cmake/CheckCopyrightScript.cmake [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                ${INPUT}
+        WORKING_DIRECTORY ${SOURCE_DIR}
+        RESULT_VARIABLE HAS_CHANGES
+        OUTPUT_QUIET)
+
+    # If the file has changes by a LANL email, the copyright must be updated
+    if (HAS_CHANGES)
+        set(REQUIRES_COPYRIGHT_UPDATE ON)
+    endif()
+endif()
```
Comment (JoshuaSBrown) [inline cmake/CheckCopyrightScript.cmake]: I don't understand this logic, if the file has not been modified why do we need the else statement?

**Comment by JoshuaSBrown** [comment]
Maybe we should also implement a target that will automatically update files that have been modified.

File: cmake/CheckCopyrightScript.cmake (hunk 2 above)
Comment (AndrewGaspar) [reply cmake/CheckCopyrightScript.cmake]: Ah, good point...

File: cmake/CheckCopyrightScript.cmake (hunk 3 above)
Comment (AndrewGaspar) [reply cmake/CheckCopyrightScript.cmake]: This was basically to catch all the cases that had slipped through before this change... I suppose we shouldn't need it anymore.

**Review by JoshuaSBrown — APPROVED** [review]
LGTM, thanks Andrew!

**Pull request merged**